### ISSUE TITLE
Add operator-pending mappings, if available

### ISSUE THREAD

**Issue by amerlyq** [albfan/ag.vim#23]
Using word under cursor is cool, but I used to search highly messed symbol combinations.
Visual highlight is good option for this, but... one key less is even better?
With guards it would look like in [vim-quickhl](https://github.com/t9md/vim-quickhl/blob/master/plugin/quickhl.vim#L107).
However, there already exist plugins for something similar, so need some discussion. You can use them for copypast/inspiration or to conclude this best to be standalone. I considered to implement this myself, but those irritative slow reactions on pull requests in main fork... [vim-operator-gsearch](https://github.com/rgrinberg/vim-operator-gsearch) and [vim-operator-search](https://github.com/osyo-manga/vim-operator-search)

**Comment by albfan** [comment]
Love the idea. Since this is intented to be a TDD project (it has a test suite but develop is not driven by test yet ), write a failed test about this feature and I will implement it on a while. 

I use [vader](https://github.com/junegunn/vader.vim) to test, but if you are new or uncomfortable with it write a pseudotest or a bash script to test it. Ask any question about that and I will guide you.

I recently write a test suite using only bash for my nerdtree-git-plugin, and is plain to understand (I guess)

https://github.com/albfan/nerdtree-git-plugin

Please add travis ci continuous integration too (if you cannot handle this I will do, but is a simple and cool task if you want to do it)

**Comment by albfan** [comment]
travis is already added. So you can see how test suite works

https://travis-ci.org/albfan/ag.vim

try to define a failed test about pending operators (although you cannot finish it, when I do it you can resolve your doubts by comparison)

Here is a pseudotest

```
""" operator-pending mappings
"""" failed

Given
  public void my_complex_function(Arg my_complex_arg) {
  ...
  }
Execute
  <AgGroupOperator>ft)
Expect
  my/file
  34:8  my_complex_function(Arg my_complex_arg) {
```

**Comment by albfan** [comment]
I'm basing the research for this feature on 

https://github.com/kana/vim-operator-user

Say we have this operators:

```
map g/ <Plug>(operator-ag) " 
```

then we want to been able to do

```
g/w
g/i)
g/iw
g/aB
```

Right? Can you define a fixture (or reuse existing) and create a failed test for this features? Expected results can be obtained with visual selection and ag shortcut.

**Comment by amerlyq** [comment]
Alright, I squashed my previous branch into master. 
It also was blocked by feature_args_derive, but I decided to implement naive function selection for now to make commits more independent. Therefore you can try it out already, as it was developed long time ago. Mappings are temporary (and bad) variant, but with them manual testing is easier.
Tests will be added soon, as all necessary pieces I already have merged into master.

Now it seems like all other mappings can be specified through this operator.
Like mapping for `cword` is actually `<Plug>(operator-ag-qf-do)iw`.
And this way is much more flexible.
Then if you want to make permanent dependence on operator-pending, we can remove many manual GetArgs analogues, etc...
I don't see much difficulties to install vim-operator-user if you have already installed ag.vim.
What you think?
If not, we can provide main (operator-pending) mappings and backup it by fallback functionality to use our functions if operator-pending is not available.
However it will be code duplication in my thinking. 
But if you think it will make merging such radical changes in upstream easier...

P.S. Why you haven't disabled appveyor yet?

**Comment by albfan** [comment]
Appveyor should be closed by now. Not much time this day, I will check all this work as soon as possible
